iidm-ddb/iidm-ddb-eurostag-import-export/src/main/java/eu/itesla_project/iidm/ddb/eurostag_imp_exp/DynamicDatabaseMock.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2017, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package eu.itesla_project.iidm.ddb.eurostag_imp_exp;
+
+import com.google.common.base.Strings;
+import com.google.common.io.ByteStreams;
+import com.google.common.io.CharStreams;
+import eu.itesla_project.iidm.network.Bus;
+import eu.itesla_project.iidm.network.Generator;
+import eu.itesla_project.iidm.network.Network;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * @author Christian Biasuzzi <christian.biasuzzi@techrain.it>
+ */
+class DynamicDatabaseMock implements DynamicDatabaseClient {
+
+    private static final String MINIMAL_DTA_TEMPLATE = "/sim_min.dta";
+    private static List<String> MOCK_REG_FILES_PREFIXES = Arrays.asList("dummefd", "dummycm");
+    private static List<String> REG_EXTENSIONS = Arrays.asList("fri", "frm", "par", "pcp", "rcp");
+
+    static Logger LOGGER = LoggerFactory.getLogger(DynamicDatabaseMock.class);
+
+    @Override
+    public void dumpDtaFile(Path workingDir, String fileName, Network network, Map<String, Character> parallelIndexes, String eurostagVersion, Map<String, String> iidm2eurostagId) {
+        Objects.requireNonNull(workingDir);
+        Objects.requireNonNull(fileName);
+        Objects.requireNonNull(network);
+        Objects.requireNonNull(parallelIndexes);
+        Objects.requireNonNull(eurostagVersion);
+        Objects.requireNonNull(iidm2eurostagId);
+
+        //uses the first connected generator that is available in the iidm2eurostag map
+        Generator generator = network.getGeneratorStream().filter(gen -> ((iidm2eurostagId.containsKey(gen.getId())) && (gen.getTerminal().isConnected()))).findFirst().get();
```

Comment:
The findFirst() method return an optional. Call the get() method can throw a NoSuchElementException. You should use orElse(null) instead